rad/src/operators/string.rs

```diff
@@ -35,6 +36,103 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
     item.try_into()
 }
 
+fn magic_map_cleaning(input: BTreeMap<String, RadonTypes>) -> BTreeMap<String, RadonTypes> {
+    let mut map = input;
+    let label = match map.remove("_label") {
+        Some(RadonTypes::String(rad_string)) => rad_string.value(),
+        _ => unreachable!("XML parsing error"),
+    };
+
+    match map.remove("_children") {
+        Some(RadonTypes::Array(rad_array)) => {
+            let children = rad_array.value();
+            let children_len = children.len();
+
+            match (map.is_empty(), children_len) {
+                (true, x) if x == 0 => {
+                    map.insert(label, RadonString::from("".to_string()).into());
+                }
+                (true, x) if x == 1 => {
+                    map.insert(label, children.first().unwrap().clone());
+                }
+                (false, x) if x == 0 => {
+                    map.insert("_label".to_string(), RadonString::from(label).into());
+                }
+                _ => {
+                    let mut text_counter = 0;
+                    let mut new_children = vec![];
+                    for child in children {
+                        match child {
+                            RadonTypes::String(rad_string) => {
+                                map.insert(format!("_text#{}", text_counter), rad_string.into());
+                                text_counter += 1;
+                            }
+                            RadonTypes::Map(rad_map) if rad_map.value().len() == 1 => {
+                                let (k, v) = rad_map.value().into_iter().next().unwrap();
+                                map.insert(k, v);
+                            }
+                            x => new_children.push(x),
+                        }
+                    }
+                    map.insert("_label".to_string(), RadonString::from(label).into());
+                    if !new_children.is_empty() {
+                        map.insert(
+                            "_children".to_string(),
+                            RadonArray::from(new_children).into(),
+                        );
+                    }
+                }
+            }
+        }
+        _ => unreachable!("XML parsing error"),
+    }
+
+    map
+}
+
+pub fn parse_element_map(input: minidom::Element, use_namespace: bool) -> RadonMap {
+    let mut map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+    map.insert("_label".to_string(), RadonString::from(input.name()).into());
+    if use_namespace {
+        map.insert("_xmlns".to_string(), RadonString::from(input.ns()).into());
+    }
+    for (k, v) in input.attrs() {
+        map.insert(format!("@{}", k), RadonString::from(v).into());
+    }
+
+    let mut a = vec![];
+    for children in input.nodes() {
+        match children {
+            minidom::Node::Element(elem) => {
+                let radon_map = parse_element_map(elem.clone(), false);
```

Comment:
You can avoid this clone by making `parse_element_map` take a `&minidom::Element`.